xds/internal/client/xds.go

```diff
@@ -278,6 +281,64 @@ func processServerSideListener(lis *v3listenerpb.Listener) (*ListenerUpdate, err
 	return lu, nil
 }
 
+func validateNetworkFilterChains(filterChains []*v3listenerpb.FilterChain) error {
+	for _, filterChain := range filterChains {
+		if filterChain == nil {
+			continue
+		}
+		seenNames := make(map[string]bool, len(filterChain.GetFilters()))
+		seenHCM := false
+		for _, filter := range filterChain.GetFilters() {
+			name := filter.GetName()
+			if name == "" {
+				return errors.New("filter missing name field")
```

Comment:
Can we improve this error message, and the ones below, at all?  (e.g. there is a name field in the FilterChain this filter is within, although it seems optional.)

Reply:
Tried adding the `filter_chain` and the `filter` wherever appropriate. let me know if this looks better.